api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -763,9 +765,26 @@ public User setUserActivationKey(User user) {
 		LoginCredential credentials = dao.getLoginCredential(user);
 		credentials.setActivationKey(activationKey);	
 		dao.setUserActivationKey(credentials);	
-			
-		//Send Email with unhashed  activation key and Date:
 		
+		String link = Context.getAdministrationService().getGlobalProperty(OpenmrsConstants.GP_HOST_URL) + token;
```

Comment:
this is not correct, the URL should have a placeholder to replace the token, you should not be appending it

Reply:
Ok